### DISCUSSION TITLE
"Field" Setting on form component missing options

### DISCUSSION THREAD

**Discussion by ndlang** [Budibase/budibase#4342] [Help] [answered]
I generated a screen for the Member table in my data connection and when adjusting it I found that one of the relationship fields "company_id" did not show up in the 'Field" options:
![image](https://user-images.githubusercontent.com/17033357/152651107-7eefb7aa-2e34-4a7b-96d7-c36c71f35254.png)
generated field:
![image](https://user-images.githubusercontent.com/17033357/152651136-4f454db9-9fd7-4fb5-b32e-97adf71feadd.png)

Couple other questions:
1. I am using a couple different Data Providers inside the Form >>Field Group and the layout seems to be affected i.e:
![image](https://user-images.githubusercontent.com/17033357/152651414-97948643-be67-43e8-a04c-50a87be28cca.png)
  - If i move the Data provider outside the Form it looks to work correctly ie
![image](https://user-images.githubusercontent.com/17033357/152651469-21e2fa61-19c8-42af-8b31-ab887620e569.png)

How would I get the DP at the same level or adjust the layout to be consistent?

**Comment by mjashanks** [comment, marked as answer]
Hi Nathan.

### Relationship Field

My guess is that your generated field is actually a `Number Field` - but you are trying to add a `Text Field` component.

It's also worth noting that I you are using a SQL datasource, you can setup your relationships, so that Budibase will generate a "Relationship Picker" for you - i.e. you can choose the related record from a dropdown.

![image](https://user-images.githubusercontent.com/3524181/152774627-30e08be4-5cdd-46d6-a0d7-948eb33cadcd.png)


### Dataprovider / Options picker layout

I've run in to this as well, so I know what you mean. The ways to get around this are to either

- position all your input labels on "Top" - this is done in the Field Group component
![image](https://user-images.githubusercontent.com/3524181/152775013-181e4a36-2e31-4b72-872e-760eddf72960.png)

- Put your data provider outside of your field group / form. This is not ideal as it means that everything is nested underneath it.

**Comment by ndlang** [comment]
Hey mjashanks,

Thank you for the quick reply. 
1. For the MySQL relationships, doesn't that create a field in the source system?
2. What if I have multiple data providers:
    - I have 3 that need to populate different fields

**Reply by mjashanks** [reply]
1. No, no columns are create in the database. It applied a `JOIN` in the select query, and pulls back the "Display Column" from the joined table as an Alias. 
e.g. 
```sql
SELECT customers.*, contacts.name As Contact
FROM customers 
INNER JOIN contacts on contacts.customerId = customers.Id
```

2. Then I would avoid nesting them all inside each other. Just put your field labels on top, and keep your structure as is.
